core/src/main/scala/org/apache/spark/util/JavaModuleUtils.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.util
+
+import scala.collection.mutable
+
+import org.apache.commons.lang3.{JavaVersion, SystemUtils}
+
+import org.apache.spark.SparkConf
+import org.apache.spark.internal.config.{DRIVER_JAVA_OPTIONS, EXECUTOR_JAVA_OPTIONS, OptionalConfigEntry}
+
+object JavaModuleUtils {
+
+  private val javaModuleOptions = Set(
+    "--add-opens=java.base/java.lang=ALL-UNNAMED",
```

Comment:
The problem is, if these are necessary to work on JDK 17, then Spark doesn't work at all without these right? that's not great, as most users won't know or want to set this up in their own execution. Is there any other workaround, or does JDK 17 now enforce this restriction?

Reply:
Yes，Spark doesn't work without these. It seems that JDK 17 enforce this restriction now. I haven't found a workaround way to prevent these exceptions from being thrown. Let me investigate again.

Reply:
Yes, @srowen . This has been warning at Java 11 ~ 16, and Java 17 enforces it finally.

Reply:
@dongjoon-hyun @srowen Or can we configure a comprehensive set of `--add-opens` args for users by default (not only for UTs), and add `-XX:+IgnoreUnrecognizedVMOptions`  at the same time? maybe this can reduce the use cost with JDK 17 of users.

Reply:
> that's not great, as most users won't know or want to set this up in their own execution. Is there any other workaround, or does JDK 17 now enforce this restriction?

The long-term solution is to not rely on Java-internal APIs anymore; they will be changed/removed more frequently, and with less notice, in the future; there's now an [explicit](https://www.youtube.com/watch?v=LgnZ-wp3D80&t=1023s) "please, do not use them". In other words, `--add-opens` is a stop gap solution only.

Reply:
Right, that is clear. Take the new memory access API. We can use it, but can't only use it until Java 14 support is dropped, and that's far away. It's either support two approaches at once now, or use these flags, and the latter seems much simpler still for now. So the question is mostly how to make that least surprising for users.